src/svg/qsvggenerator.cpp

```diff
@@ -83,6 +85,63 @@ static void translate_dashPattern(const QVector<qreal> &pattern, qreal width, QS
     pattern_string->chop(1);
 }
 
+// Converts a QPainterPath to the contents of the 'd' attribute on an svg path start tag
+static QString qPainterPathToPathData(const QPainterPath &p) {
+    QString path;
+    QTextStream path_stream(&path);
+
+    for (int i = 0; i < p.elementCount(); ++i) {
+        const QPainterPath::Element &e = p.elementAt(i);
+        switch (e.type) {
+        case QPainterPath::MoveToElement:
+            path_stream << 'M' << e.x << ',' << e.y;
+            break;
+        case QPainterPath::LineToElement:
+            path_stream << 'L' << e.x << ',' << e.y;
+            break;
+        case QPainterPath::CurveToElement:
+            path_stream << 'C' << e.x << ',' << e.y;
+            ++i;
```

Comment:
tbh this code is inherited from the original Qt code with no modifications, the diff is just because I moved it (after Michael also originally moved it).

I do find it a bit confusing to understand, but I don't know if it's worth me changing or not?

Reply:
ok, if it wasn't written by you, it is okay, let's keep as it is for sake of making migration to next QT version easier.